diff --git a/Distribution/Simple/GHCJS.hs b/Distribution/Simple/GHCJS.hs
index 46896c3a0..90a0411b3 100644
--- a/Distribution/Simple/GHCJS.hs
+++ b/Distribution/Simple/GHCJS.hs
@@ -947,6 +947,7 @@ decodeMainIsArg arg
 -- | A collection of:
 --    * C input files
 --    * C++ input files
+--    * JS input files
 --    * GHC input files
 --    * GHC input modules
 --
@@ -954,6 +955,7 @@ decodeMainIsArg arg
 data BuildSources = BuildSources {
         cSourcesFiles      :: [FilePath],
         cxxSourceFiles     :: [FilePath],
+        jsSourceFiles      :: [FilePath],
         inputSourceFiles   :: [FilePath],
         inputSourceModules :: [ModuleName]
     }
@@ -999,6 +1001,7 @@ gbuildSources verbosity specVer tmpDir bm =
              return BuildSources {
                         cSourcesFiles      = cSources bnfo,
                         cxxSourceFiles     = cxxSources bnfo,
+                        jsSourceFiles      = jsSources bnfo,
                         inputSourceFiles   = [main],
                         inputSourceModules = filter (/= mainModName) $ exeModules exe
                     }
@@ -1006,6 +1009,7 @@ gbuildSources verbosity specVer tmpDir bm =
           else return BuildSources {
                           cSourcesFiles      = cSources bnfo,
                           cxxSourceFiles     = cxxSources bnfo,
+                          jsSourceFiles      = jsSources bnfo,
                           inputSourceFiles   = [main],
                           inputSourceModules = exeModules exe
                       }
@@ -1019,6 +1023,7 @@ gbuildSources verbosity specVer tmpDir bm =
              in  return BuildSources {
                             cSourcesFiles      = csf,
                             cxxSourceFiles     = cxxsf,
+                            jsSourceFiles      = [],
                             inputSourceFiles   = [],
                             inputSourceModules = exeModules exe
                         }
@@ -1028,6 +1033,7 @@ gbuildSources verbosity specVer tmpDir bm =
         BuildSources {
             cSourcesFiles      = cSources bnfo,
             cxxSourceFiles     = cxxSources bnfo,
+            jsSourceFiles      = [],
             inputSourceFiles   = [],
             inputSourceModules = foreignLibModules flib
         }
@@ -1082,6 +1088,7 @@ gbuild verbosity numJobs pkg_descr lbi bm clbi = do
 
   let cSrcs               = cSourcesFiles buildSources
       cxxSrcs             = cxxSourceFiles buildSources
+      jsSrcs              = jsSourceFiles buildSources
       inputFiles          = inputSourceFiles buildSources
       inputModules        = inputSourceModules buildSources
       isGhcDynamic        = isDynamic comp
@@ -1134,8 +1141,8 @@ gbuild verbosity numJobs pkg_descr lbi bm clbi = do
                                                 PD.frameworks bnfo,
                       ghcOptLinkFrameworkDirs = toNubListR $
                                                 PD.extraFrameworkDirs bnfo,
-                      ghcOptInputFiles     = toNubListR
-                                             [tmpDir </> x | x <- cObjs ++ cxxObjs]
+                      ghcOptInputFiles     = toNubListR $
+                                             [tmpDir </> x | x <- cObjs ++ cxxObjs] ++ jsSrcs
                     }
       dynLinkerOpts = mempty {
                       ghcOptRPaths         = rpaths
